### PR TITLE
[V1][TPU] Apply the ragged paged attention kernel fix and remove the padding.

### PR DESCRIPTION
**Pull request by vanbasten23**
In https://github.com/vllm-project/vllm/pull/14597, we added the padding due to an ragged paged attention kernel issue. Since we have fixed the issue in the kernel, we can remove the padding in vLLM. This will make the kernel run more efficiently.

Test plan:
1. [Repro script](https://gist.github.com/vanbasten23/f00868169d6e4e68edb34d99ee7f287e) for https://github.com/vllm-project/vllm/pull/14597
2. VLLM_USE_V1=1 pytest -s -v vllm/tests/entrypoints/llm/test_accuracy.py::test_lm_eval_accuracy_v1_engine 2>&1 | tee out.txt
3.
```
MODEL=meta-llama/Meta-Llama-3.1-8B
XLA_HLO_DEBUG=1 VLLM_USE_V1=1 vllm serve $MODEL --seed 42 --disable-log-requests --download_dir "/mnt/disks/persist" --gpu-memory-utilization 0.95 --max-num-batched-tokens 512 --max-num-seqs 512 --tensor-parallel-size 1 --max-model-len 2048 &
XLA_HLO_DEBUG=1 VLLM_USE_V1=1 python3 benchmarks/benchmark_serving.py --backend vllm --model $MODEL --dataset-name random --random-input-len 1700 --random-output-len 128 --random-prefix-len 0 --num-prompts 50
```

### CODE REVIEW

File: vllm/v1/worker/tpu_model_runner.py [hunk 1]
```diff
@@ -139,10 +138,8 @@ def __init__(
                                             device="cpu")
         self.slot_mapping_np = self.slot_mapping_cpu.numpy()
 
-        padded_max_num_blocks_per_req = _get_padded_number(
-            self.max_num_blocks_per_req, NUM_KV_PAGES_PER_BLOCK)
         self.block_table_cpu = torch.zeros(
-            (self.max_num_tokens, padded_max_num_blocks_per_req),
+            (self.max_num_tokens, self.max_num_blocks_per_req),
```
Comment (robertgshaw2-redhat) [inline vllm/v1/worker/tpu_model_runner.py:142]: What are the contraints of the kernel now? Multiple of 2?

Comment (vanbasten23) [reply vllm/v1/worker/tpu_model_runner.py:142]: block_table.shape[1] doesn't have any constraint now.

**Review by mgoin — APPROVED** [review]

File: vllm/v1/worker/tpu_model_runner.py (hunk 1 above)
Comment (mgoin) [reply vllm/v1/worker/tpu_model_runner.py:142]: Do we not need any changes to the slicing logic into this buffer during execution?
